data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostValidatorLiveness.java

```diff
@@ -13,73 +13,99 @@
-import tech.pegasys.teku.api.request.v1.validator.ValidatorLivenessRequest;
+import tech.pegasys.teku.api.migrated.ValidatorLivenessAtEpoch;
+import tech.pegasys.teku.api.migrated.ValidatorLivenessRequest;
 import tech.pegasys.teku.api.response.v1.validator.PostValidatorLivenessResponse;
-import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
-import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
 
-public class PostValidatorLiveness extends AbstractHandler {
-  private static final Logger LOG = LogManager.getLogger();
+public class PostValidatorLiveness extends MigratingEndpointAdapter {
   public static final String ROUTE = "/eth/v1/validator/liveness";
 
+  private static final SerializableTypeDefinition<List<ValidatorLivenessAtEpoch>> RESPONSE_TYPE =
+      SerializableTypeDefinition.<List<ValidatorLivenessAtEpoch>>object()
+          .name("PostValidatorLivenessResponse")
+          .withField(
+              "data", listOf(ValidatorLivenessAtEpoch.getJsonTypeDefinition()), Function.identity())
+          .build();
+
   public final ChainDataProvider chainDataProvider;
   public final NodeDataProvider nodeDataProvider;
   public final SyncDataProvider syncDataProvider;
 
+  public PostValidatorLiveness(final DataProvider provider) {
+    this(
+        provider.getChainDataProvider(),
+        provider.getNodeDataProvider(),
+        provider.getSyncDataProvider());
+  }
+
   public PostValidatorLiveness(
       final ChainDataProvider chainDataProvider,
       final NodeDataProvider nodeDataProvider,
-      final SyncDataProvider syncDataProvider,
-      final JsonProvider jsonProvider) {
-    super(jsonProvider);
+      final SyncDataProvider syncDataProvider) {
+    super(
+        EndpointMetadata.post(ROUTE)
+            .operationId("postValidatorLiveness")
+            .summary("Get Validator Liveness")
+            .description(
+                "Requests the beacon node to indicate if a validator has been"
+                    + "    observed to be live in a given epoch. The beacon node might detect liveness by"
+                    + "    observing messages from the validator on the network, in the beacon chain,"
+                    + "    from its API or from any other source. It is important to note that the"
+                    + "    values returned by the beacon node are not canonical; they are best-effort"
+                    + "    and based upon a subjective view of the network.")
```

Comment:
just remove the extra white space, either put 1 space at end of line and its neat here, or 1 space at start... dealers choice